### ISSUE TITLE
lag on vote starts/end latest updates csgo

### ISSUE THREAD

**Issue by Ayrton09** [Silenci0/UMC#12]
hello silence, i dont know if my problem or something but sometimes when vote starts or end the server get a big spike lag and sometimes time out for some seconds. i tried with sm 1.9 and 1.10 and the problem persists.

**Comment by Silenci0** [comment]
Hello Ayrton09,

Thank you for bringing this to my attention. Many users have reported lag from this plugin both here and in the UMC thread in the past as well, so I don't believe it is just your issue alone. In general, it seemed to be one (or more) of the following that contributed to the lag:

- Large map cycle lists, particularly those of 100+ maps in them and/or combined with other features in the plugin.
- Use of the Display option for map names. This in particular was a problem with larger map cycle files.
- Too many groups and/or duplicate maps in across group. This is not always the root issue, but it can contribute to it.

Judging from the sound of your issue, since its happening when you start and end a vote, it is most likely something to do with what it is trying to display, which might mean there are too many maps or there is too much complexity with map options. In general, reducing the number of maps/groups and/or not using the Display option can reduce lag issues. 

With all that said, it is also possible that a CS:GO update might have helped cause this issue as well or the update helped to make certain plugin-based issues worse. From what I can see, the devs fixed an issue with vote commands, so that might be one cause of the problem, though UMC doesn't necessarily use that system since it has its own.

**Comment by Ayrton09** [comment]
thanks for your answer. particularly i don't have display option, no duplicate maps and have the same amount of maps since months so is weird but i will try to remove some maps or something.

**Comment by Silenci0** [comment]
I see, that is pretty strange. You might be right about the issue being caused by the latest CS:GO update, so I'll label this specifically for CS:GO as a bug. Thank you for the report!

**Comment by Silenci0** [comment]
Looking further into this, I tested the latest version of UMC (3.7.0) in my CS:GO test server and I did not run into any lag when the map vote was displayed. I did this for several rounds and did not run into any issues with lag as a result of the voting, RTV, and/or nominations menu. 

That said, my test server is pretty bare bones and does not have a lot of plugins (only uses stock Sourcemod plugins + UMC and SMAC), only uses vanilla/stock maps, and only runs bots (no players can get into this server). This might drastically affect the testing of this plugin when searching for this issue, but so far I cannot replicate this behavior. 

I'm not sure if this issue is still occurring for you or not, but if it is, try using plugins from UMC 3.7.0 and see if it might resolve those issues. If it still occurs, the issue might be someplace else (such as plugins or server settings) that might be an issue.
